src/main/java/org/bytecodeparser/instruction/Instruction.java

```diff
@@ -2,11 +2,24 @@
 
 import org.bytecodeparser.print.CustomPrettyPrint;
 
+import java.io.DataInputStream;
+import java.util.Arrays;
+
 @CustomPrettyPrint
-public interface Instruction {
+public class Instruction {
+    final InstructionArguments args;
+    private final InstructionTypes type;
+
+    public Instruction(Integer opCode, DataInputStream dataInputStream) {
+        this.type = Arrays.stream(InstructionTypes.values())
+                .filter(instructionTypes -> opCode.equals(instructionTypes.opCode))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("Unable to read instruction with opcode = " + opCode));
```

Comment:
everytime Instruction is created - there is new stream created - it is inefficient - you can factor out it to the static Map where Key is opcode and the value is InstructionType instance